app/controllers/users_controller.rb

```diff
@@ -1,43 +1,62 @@
 class UsersController < ApplicationController
   def index
+    if @login_user.nil?
```

Comment:
This would be better done with a [controller filter.](https://guides.rubyonrails.org/action_controller_overview.html#filters)